sklearn/linear_model/_glm/glm.py

```diff
@@ -267,6 +963,35 @@ def fit(self, X, y, sample_weight=None):
             )
             self.n_iter_ = _check_optimize_result("lbfgs", opt_res)
             coef = opt_res.x
+        elif self.solver in ["newton-cholesky", "newton-qr-cholesky"]:
+            sol_dict = {
+                "newton-cholesky": CholeskyNewtonSolver,
+                "newton-qr-cholesky": QRCholeskyNewtonSolver,
+            }
+            sol = sol_dict[self.solver](
+                coef=coef,
+                linear_loss=linear_loss,
+                l2_reg_strength=l2_reg_strength,
+                tol=self.tol,
+                max_iter=self.max_iter,
+                n_threads=n_threads,
+                verbose=self.verbose,
+            )
+            coef = sol.solve(X, y, sample_weight)
+            self.n_iter_ = sol.iteration
+        elif issubclass(self.solver, NewtonSolver):
+            sol = self.solver(
+                coef=coef,
+                linear_loss=linear_loss,
+                l2_reg_strength=l2_reg_strength,
+                tol=self.tol,
+                max_iter=self.max_iter,
+                n_threads=n_threads,
+            )
+            coef = sol.solve(X, y, sample_weight)
+            self.n_iter_ = sol.iteration
+        else:
+            raise TypeError(f"Invalid solver={self.solver}.")
```

Comment:
This is unreachable code
```suggestion
```

Reply:
Why? Technically it could happen with https://github.com/scikit-learn/scikit-learn/pull/23314/files#diff-60d20f9c009b397dbfd80c5b38734be5195569042501980b7231104b91662147R823.

Reply:
Ah I see, because of ``Hidden(type),``.

Reply:
Yes.